Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard_test.py

```diff
@@ -0,0 +1,264 @@
+create_grade_alert_mock = load_json("./test_data/alerts/create_grade_alert.json")
+create_score_alert_mock = load_json("./test_data/alerts/create_score_alert.json")
+services_mock = load_json("./test_data/companies/services.json")
+
+""" Helper Functions Unit Tests"""
+
+
+def test_incidents_to_import(mocker):
+
+    mocker.patch.object(client, "get_alerts_last_week", return_value=alerts_mock)
+
+    response = client.get_alerts_last_week(email="some@email.com")
+
+    assert response.get('entries')
+
+    entries = response.get('entries')
+
+    assert len(entries) == 9
+
+    # 3 day in seconds
+    seconds_ago = 3 * 86400
+
+    # Set runtime
+    now = int(datetime.datetime(2021, 7, 12).timestamp()) - seconds_ago
+
+    incidents = incidents_to_import(entries)
+
+    assert len(incidents) == 0
+
+    # Iterate over each incident and ensure they
+    # were supposed to be imported
+    for incident in incidents:
+        incident_time = incident["occurred"]
+        incident_timestamp = int(datetime.datetime.strptime(incident_time, DATE_FORMAT).timestamp())
+        assert incident_timestamp > now
+
+
+""" TEST CONSTANTS """
+
+
+USERNAME = "user@domain.com"
+PORTFOLIO_ID = "1"
+PORTFOLIO_ID_NE = "2"
+DOMAIN = "domain1.com"
+DOMAIN_NE = "domain2.com"
+ALERT_ID = "1"
+ALERT_ID_NE = "2"
+
+""" Endpoints """
+
+PORTFOLIO_ROOT_EP = "/portfolios"
+PORTFOLIO_FOUND_EP = "{0}/1/companies".format(PORTFOLIO_ROOT_EP)
+PORTFOLIO_FOUND_EP = "{0}/2/companies".format(PORTFOLIO_ROOT_EP)
+
+COMPANIES_ROOT_EP = "/companies"
+COMPANIES_SCORE_EP = "{0}/{1}".format(COMPANIES_ROOT_EP, DOMAIN)
+COMPANIES_FACTOR_SCORE_EP = "{0}/{1}/factors".format(COMPANIES_ROOT_EP, DOMAIN)
+COMPANIES_HISTORY_SCORE_EP = "{0}/{1}/history/score".format(COMPANIES_ROOT_EP, DOMAIN)
+COMPANIES_HISTORY_FACTOR_SCORE_EP = "{0}/{1}/history/factor/score".format(COMPANIES_ROOT_EP, DOMAIN)
+COMPANIES_SERVICES_EP = "{0}/{1}/services".format(COMPANIES_ROOT_EP, DOMAIN)
+
+ALERTS_ROOT_EP = "/users/by-username/{0}/alerts".format(USERNAME)
+ALERTS_TYPE_SCORE = "score"
+ALERTS_TYPE_GRADE = "grade"
+ALERTS_GRADE_EP = "{0}/grade".format(ALERTS_ROOT_EP)
+ALERTS_SCORE_EP = "{0}/score".format(ALERTS_ROOT_EP)
+ALERTS_DELETE_SCORE_EP = "{0}/{1}/{2}".format(ALERTS_ROOT_EP, ALERTS_TYPE_SCORE, ALERT_ID)
+ALERTS_DELETE_GRADE_EP = "{0}/{1}/{2}".format(ALERTS_ROOT_EP, ALERTS_TYPE_GRADE, ALERT_ID)
+NOTIFICATIONS_ROOT_EP = "/users/by-username/{0}/notifications/recent".format(USERNAME)
+
+
+""" Client Unit Tests """
+
+
+MOCK_URL = "mock://securityscorecard-mock-url"
+
+client = Client(
+    base_url=MOCK_URL,
+    verify=False,
+    proxy=False
+)
+
+
+def test_portfolios_list(mocker):
+
+    mocker.patch.object(client, "get_portfolios", return_value=portfolios_mock)
+
+    response = client.get_portfolios()
+
+    assert response == portfolios_mock
+
+
+def test_portfolio_list_companies(mocker):
+
+    """
+        Checks cases where the portfolio exists and doesn't exist
+    """
```

Comment:
Use the 'Given','when','then' model. See NetscoutArborSightline if you need example